jgrapht-core/src/main/java/org/jgrapht/generate/NamedGraphGenerator.java

```diff
@@ -0,0 +1,1033 @@
+        for(int[] edge : edges)
+            addEdge(targetGraph, edge[0], edge[1]);
+    }
+
+    //-------------Thomsen Graph-----------//
+    /**
+     * @see #generateThomsenGraph
+     * @return Thomsen Graph
+     */
+    public static Graph<Integer, DefaultEdge> thomsenGraph(){
+        Graph<Integer, DefaultEdge> g=new SimpleGraph<>(DefaultEdge.class);
+        new NamedGraphGenerator<Integer, DefaultEdge>(new IntegerVertexFactory()).generateThomsenGraph(g);
+        return g;
+    }
+    /**
+     * Generates the <a href="http://mathworld.wolfram.com/UtilityGraph.html">Thomsen Graph</a>.
+     * The Thomsen Graph is complete bipartite graph consisting of 6 vertices (3 vertices in each bipartite
+     * partition. It is also called the Utility graph.
+     * @param targetGraph receives the generated edges and vertices; if this is non-empty on entry, the
+     *        result will be a disconnected graph since generated elements will not be connected to
+     *        existing elements
+     */
+    public void generateThomsenGraph(Graph<V,E> targetGraph){
+        vertexMap.clear();
+        int[][] edges={{0,3},{0,4},{0,5},{1,3},{1,4},{1,5},{2,3},{2,4},{2,5}};
+        for(int[] edge : edges)
+            addEdge(targetGraph, edge[0], edge[1]);
+    }
+
+    //--------------Helper methods-----------------/
+    private V addVertex(Graph<V,E> targetGraph, int i){
+        if(!vertexMap.containsKey(i)) {
+            V v=vertexFactory.createVertex();
+            vertexMap.put(i, v);
+            targetGraph.addVertex(v);
+        }
+        return vertexMap.get(i);
+    }
+    private void addEdge(Graph<V,E> targetGraph, int i, int j){
+        V u = addVertex(targetGraph, i);
+        V v = addVertex(targetGraph, j);
+        targetGraph.addEdge(u, v);
+    }
+
+    /**
+     * Creates vertices of the type Integer
+     */
+    public static class IntegerVertexFactory implements VertexFactory<Integer>
```

Comment:
Forgot one comment:  this is the third copy of this class (the other two are in tests).  Probably time to make it official as a top-level class in org.jgrapht.alg.util?